src/main/java/com/vmware/burp/extension/service/BurpService.java

```diff
@@ -353,13 +357,43 @@ public void excludeFromScope(String urlString) throws MalformedURLException {
         return scanIssues;
     }
 
-    public byte[] generateScanReport(String urlPrefix, ReportType reportType) throws IOException {
+    public byte[] generateScanReport(String urlPrefix, ReportType reportType, IssueSeverity[] issueSeverities,
+                                     IssueConfidence[] issueConfidences, String[] issuePaths) throws IOException {
         Path reportFile = Files.createTempFile("Report", "." + reportType.getReportType());
         reportFile.toFile().deleteOnExit();
-        LegacyBurpExtender.getInstance().getCallbacks()
-                .generateScanReport(reportType.getReportType(),
-                        LegacyBurpExtender.getInstance().getCallbacks().getScanIssues(urlPrefix),
-                        reportFile.toFile());
+
+        IBurpExtenderCallbacks burpExtenderCallbacks = LegacyBurpExtender.getInstance().getCallbacks();
+
+        Supplier<Stream<IScanIssue>> scanIssuesSupplier = () -> Arrays.stream(burpExtenderCallbacks.getScanIssues(urlPrefix));
```

Comment:
I am not sure it makes sense to have _urlPrefix_ as well as _issuePath_

_urlPrefix_  is actually a URL so in the form http(s)://doyensec.com/
_issuePath_ are just URI paths

We can leave as is, but I would adjust the description in the `ApiImplicitParam` to specify that for users